### DISCUSSION TITLE
Dockerize with pnpm

### DISCUSSION THREAD

**Discussion by naoyoshi** [nuxt/nuxt#21376] [Questions]
Has anyone succeeded to dockerize Nuxt.js 3 latest version with pnpm?
I'm looking for the way with pnpm, though, no information.
If anyone could do it, please tell me the way.
I'm glad that if you upload sample files to Github.

**Comment by danielvlla** [comment]
This is one way of doing this, of course you can increase complexity to get smaller image sizes by utilising multi stage builds.

```
FROM node:20-bookworm

WORKDIR /frontend

COPY . .

RUN npm install -g pnpm
RUN pnpm install --frozen-lockfile

ENV NODE_ENV=production

ENV NUXT_PORT=5000
ENV NITRO_PORT=5000
ENV NITRO_HOST=0.0.0.0
ENV NUXT_HOST=0.0.0.0
# ENV HOST=0.0.0.0

EXPOSE 5000

RUN pnpm run build
CMD ["node", "/frontend/.output/server/index.mjs"]
```

**Comment by martinszeltins** [comment]
Yes, I have dockerized it both the dev server and the production build running in a container - https://github.com/martinszeltins/nuxt-docker-cpu
